src/index.jst

```diff
@@ -2,44 +2,45 @@
 
 // do not edit .js files directly - edit src/index.jst
 
+{{? it.es6 }}
+var envHasBigInt64Array = typeof BigInt64Array !== 'undefined';
+{{?}}
+
 module.exports = function equal(a, b) {
   if (a === b) return true;
 
-  if (a && b && typeof a == 'object' && typeof b == 'object') {
+  if (a && b && typeof a ==='object' && typeof b ==='object') {
     if (a.constructor !== b.constructor) return false;
 
-    var length, i, key, keys;
+    var length, i, key;
     if (Array.isArray(a)) {
       length = a.length;
-      if (length != b.length) return false;
+      if (length !== b.length) return false;
       for (i = length; i-- !== 0;)
         if (!equal(a[i], b[i])) return false;
       return true;
     }
 
 {{? it.es6 }}
+    var keys;
     if (a instanceof Map) {
       if (a.size !== b.size) return false;
-
-      keys = getKeys(a);
-      for (i = a.size; i-- !== 0;)
-        if (!b.has(keys[i])) return false;
-
-      for (i = a.size; i-- !== 0;) {
-        key = keys[i];
-        if (!equal(a.get(key), b.get(key))) return false;
+      // should be named `iterator` for the sake of variables efficiency
+      keys = a.entries();
+      while ((i = keys.next()) && !i.done) {
+        if (!b.has(i.value[0])) return false;
+        if (!equal(i.value[1], b.get(i.value[0]))) return false;
       }
 
       return true;
     }
 
     if (a instanceof Set) {
       if (a.size !== b.size) return false;
-
-      keys = getKeys(a);
-      for (i = a.size; i-- !== 0;)
-        if (!b.has(keys[i])) return false;
-
+      // should be named `iterator` for the sake of variables efficiency
+      keys = a.entries();
+      while ((i = keys.next()) && !i.done)
```

Comment:
The same thing as with Map - reduced Set iterations.

Reply:
if we do it like this - iterator makes more sense.